File: hw/unda/U3/HW21TemplateMethod/TemplateMethodPy/model/tea.py
from model.caffeine_beverage import CaffeineBeverage

class Tea(CaffeineBeverage):
    def brew(self):
        print("Steep the tea")

    def add_condiments(self):
        print("Adding lemon")

    def wants_condiments(self):
        answer = self.get_user_input()
        return answer.lower().startswith("y")

    def get_user_input(self):
        return input("Would you like lemon with your tea (y/n)? ")

File: hw/maisincho/u3/HW21TemplateMethod/TemplateMeThod/src/main/java/ec/edu/espe/template/model/Tea.java
package ec.edu.espe.template.model;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class Tea extends CaffeineBeverage {

    @Override
    void brew() {
        System.out.println("Steep the tea");
    }

    @Override
    void addCondiments() {
        System.out.println("Adding Lemon");

    }

    @Override
    public boolean wantsCondiments() {
        String answer = getUserInput();
        if (answer.toLowerCase().startsWith("y")) {
            return true;
        } else {
            return false;
        }

    }

    private String getUserInput() {
        String answer = null;
        System.out.println("Would you like lemon on your tea (y/n) ");
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        try {
            answer = in.readLine();
        } catch (IOException e) {
            System.out.println("Error reading input " + e.getMessage());

        }
        return answer;
    }

}
